dom0/sd-export.sls

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# vim: set syntax=yaml ts=2 sw=2 sts=2 et :
+
+#
+# Installs 'sd-export' AppVM, to persistently store SD data
+# This VM has no network configured.
+##
+include:
+  - sd-workstation-template
+
+sd-export-template:
+  qvm.vm:
+    - name: sd-export-template
+    - clone:
+      - source: sd-workstation-template
+      - label: yellow
+    - tags:
+      - add:
+        - sd-workstation
+    - require:
+      - sls: sd-workstation-template
+
+sd-export-dvm:
+  qvm.vm:
+    - name: sd-export-dvm
+    - present:
+      - template: sd-export-template
+      - label: yellow
+    - prefs:
+      - netvm: ""
+      - template_for_dispvms: True
+    - tags:
+      - add:
+        - sd-workstation
+    - require:
+      - qvm: sd-export-template
+
+# Ensure the Qubes menu is populated with relevant app entries,
+# so that Nautilus/Files can be started via GUI interactions.
+sd-export-template-sync-appmenus:
+  cmd.run:
+    - name: >
+        qvm-start --skip-if-running sd-export-template &&
+        qvm-sync-appmenus sd-export-template
+    - require:
+      - qvm: sd-export-template
+    - onchanges:
+      - qvm: sd-export-template
+
+{% import_json "sd/config.json" as d %}
+
+# Here we must create as the salt stack does not appear to allow us to create
+# VMs with the class DispVM and attach the usb device specified in the config
+# permanently to this VM
+create-named-sd-export-dispvm-and-permanently-attach:
+  cmd.run:
+    - name: >
+        qvm-remove --force sd-export-usb || true;
+        qvm-create --class DispVM --template sd-export-dvm --label red sd-export-usb;
+        qvm-usb attach --persistent sd-export-usb {{ d.usb.device }} || true;
```

Comment:
It's a shame to force removal of the VM on every provisioning run. A quick glance at the qvm docs seems to confirm we don't have a Salt-managed method for this, but perhaps we could wrangle it with the Python API. Fine as-is, for now, we'll want to switch wholesale to the Python API for VM creation if necessary (#159).

Reply:
Here's a patch to improve the error handling somewhat on this rather forceful one-liner, by breaking it up:

```
Author: Conor Schaefer <conor@freedom.press>
Date:   Mon Jun 3 17:23:42 2019 -0700

    Breaks up sd-export-dvm salt config
    
    Error handling is difficult in bash one-liners, so let's leverage
    Saltstack for qvm configs as far as possible. The `qvm-check` command
    returns false if the VM does not exist, so we'll create it only
    conditionally.

diff --git a/dom0/sd-export.sls b/dom0/sd-export.sls
index 58cc269..32c879c 100644
--- a/dom0/sd-export.sls
+++ b/dom0/sd-export.sls
@@ -47,16 +47,31 @@ sd-export-template-sync-appmenus:
     - onchanges:
       - qvm: sd-export-template
 
-{% import_json "sd/config.json" as d %}
-
 # Here we must create as the salt stack does not appear to allow us to create
 # VMs with the class DispVM and attach the usb device specified in the config
 # permanently to this VM
-create-named-sd-export-dispvm-and-permanently-attach:
+sd-export-create-named-dispvm:
+  cmd.run:
+    - name: >
+        qvm-check sd-export-usb ||
+        qvm-create --class DispVM --template sd-export-usb-dvm --label red sd-export-usb
+    - require:
+      - qvm: sd-export-usb-dvm
+
+{% import_json "sd/config.json" as d %}
+
+sd-export-named-dispvm-permanently-attach-usb:
   cmd.run:
     - name: >
-        qvm-kill sd-export-usb || true;
-        qvm-remove --force sd-export-usb || true;
-        qvm-create --class DispVM --template sd-export-usb-dvm --label red sd-export-usb;
-        qvm-usb attach --persistent sd-export-usb {{ d.usb.device }} || true;
-        qvm-tags sd-export-usb add sd-workstation
+        qvm-usb attach --persistent sd-export-usb {{ d.usb.device }} || true
+    - require:
+      - cmd: sd-export-create-named-dispvm
+
+sd-export-named-dispvm-add-tags:
+  qvm.vm:
+    - name: sd-export-usb
+    - tags:
+      - add:
+        - sd-workstation
+    - require:
+      - cmd: sd-export-create-named-dispvm
```

Since I encountered problems during initial review, I'm not appending this commit, to make sure we've working from the same branch as we debug.